gui-agent/vmside.c

```diff
@@ -1680,8 +1713,16 @@ static void handle_keypress(Ghandles * g, XID UNUSED(winid))
             XFreeModifiermap(modmap);
         }
     }
-
-    feed_xdriver(g, 'K', key.keycode, key.type == KeyPress ? 1 : 0);
+        
+    if(!g->createdInputDevice) {
```

Comment:
I adjusted it now so when virtual input device creation has been requested, modifiers are synced through the uinput fd instead of the xserver fd.

They are completely separate from each other now:

if(virtual input device NOT requested){
// original handle_keypress
} else {
// new handle_keypress (only writes to the uinput fd, never to the xserver fd)
}


Before these changes, all of the modifiers worked correctly when switching between qubes, except when pressing down the modifier and switching into a qube.


Now, all of the following scenarios work as expected:

Start in qube 1
Hold down shift
Switch to qube 2
Start typing while still holding down shift
This results in uppercase characters being typed out into qube 2 as expected


Start in qube 1
Hold down shift
Switch to qube 2
Release shift
Switch back to qube 1
Start typing
This results in lowercase characters being typed out into qube 1 as expected

Open a text file in nano in qube 1
Switch to qube 2
Hold down ctrl
Switch back to qube 1
Press s while still holding down ctrl
nano receives a ctrl+s and attempts to save the file


Modifiers still work exactly as expected when not switching between qubes as well.